From 4a292721bcf2943bfc05c6a1c859992f28e3efec Mon Sep 17 00:00:00 2001
From: Ryan Lortie <desrt@desrt.ca>
Date: Wed, 25 Mar 2015 09:29:49 -0400
Subject: GListModel: roll back use of type redefinition

We declare the typedefs for GListModel and GListStore in giotypes.h, as
a matter of convention.  This is not actually required, since the
typedef is emitted as part of the G_DECLARE_* macros.

The giotypes.h approach is only used to avoid cyclic dependencies
between headers, which is not a problem in this case.

Type redefinition is a C11 feature, and although it was around in some
compilers before then, gcc 4.2.1 (from 2007) is apparently still in wide
use, being the default compiler for OpenBSD.

Eventually, we will probably hit a case where we actually need to
redefine a type, but since we're not there yet, let's back off a bit.
---
 gio/giotypes.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/gio/giotypes.h b/gio/giotypes.h
index 4342d47..53f8cc9 100644
--- a/gio/giotypes.h
+++ b/gio/giotypes.h
@@ -61,8 +61,6 @@ typedef struct _GPermission                   GPermission;
 
 typedef struct _GMenuModel                    GMenuModel;
 typedef struct _GNotification                 GNotification;
-typedef struct _GListModel                    GListModel;
-typedef struct _GListStore                    GListStore;
 
 /**
  * GDrive:
-- 
cgit v0.12

